### ISSUE TITLE
Error found when a path string in "fs" package is longer than 1024

### ISSUE THREAD

**Issue by gdsc0301** [oven-sh/bun#10200]
### What version of Bun is running?

1.1.3+2615dc742

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

The bug is that when the "fs" is used with a path string longer than 1024, it panics. It happened in this case where **esbuild** seems to get the files referenced in a css "url(path/to/file)" statement using the "fs" package, but what happens is that the "path" here is a gigantic base64 encoded image. Who would've imagine.

![Screenshot 2024-04-12 at 01 15 18](https://github.com/oven-sh/bun/assets/11581684/046c8a33-662f-47f9-99e4-263a15a19f6a)

You can reproduce my environment by creating a Remix project using TailwindCSS and Swiper, and building it using `bun run --bun dev`.

### What is the expected behavior?

The expected behaviour is the node's behavior, in Node this error doesn't happen.

### What do you see instead?

```bash
    ✘ [ERROR] Invalid path string: path is too long (max: 1024) [plugin server-bare-modules]
    node_modules/tsconfig-paths/lib/filesystem.js:4:10:
      4 │ var fs = require("fs");
        ╵           ^

    at existsSync (native)
    at fileExistsSync (dareyou.online/node_modules/tsconfig-paths/lib/filesystem.js:4:11)
    at findFirstExistingPath (dareyou.online/node_modules/tsconfig-paths/lib/match-path-sync.js:72:54)
    at matchFromAbsolutePaths (dareyou.online/node_modules/tsconfig-paths/lib/match-path-sync.js:48:9)
    at <anonymous> (dareyou.online/node_modules/tsconfig-paths/lib/match-path-sync.js:21:114)
    at resolvePath (dareyou.online/node_modules/@remix-run/dev/dist/compiler/server/plugins/bareImports.js:34:9)
    at <anonymous> (dareyou.online/node_modules/@remix-run/dev/dist/compiler/server/plugins/bareImports.js:49:21)
    at <anonymous> (dareyou.online/node_modules/@remix-run/dev/node_modules/esbuild/lib/main.js:1361:28)
    at processTicksAndRejections (:13:78)

  This error came from the "onResolve" callback registered here:

    node_modules/@remix-run/dev/dist/compiler/server/plugins/bareImports.js:12:0:
      12 │ 
         ╵ ^

    at setup (dareyou.online/node_modules/@remix-run/dev/dist/compiler/server/plugins/bareImports.js:12:24)
    at <anonymous> (dareyou.online/node_modules/@remix-run/dev/node_modules/esbuild/lib/main.js:1328:28)
    at processTicksAndRejections (:12:39)

  The plugin "server-bare-modules" was triggered by this import

    css-side-effects-plugin-ns:node_modules/swiper/swiper-bundle.css:16:11:
      16 │   src: url('data:application/font-woff;charset=utf-8;base64, d09GR...
```

### Additional information

This is a Remix project, I'm using [TailwindCSS](https://tailwindcss.com/), [SwiperReact](https://swiperjs.com/swiper-api#parameters), [Mantine UI](https://mantine.dev/getting-started/) and [Tabler Icons](https://tabler.io/icons).

**Comment by gvilums** [comment]
This is interesting - macOS has an internal maximum path length of 1024 characters, so I'm not sure how node is handling this correctly.

**Comment by gvilums** [comment]
On second glance, it looks like bun and node return different errors in the following example code. This could be the cause of this issue if some code depends on the exact error to match

**Comment by gvilums** [comment]
@gdsc0301 could you provide an example repository that I can clone to reproduce this?

**Comment by Jarred-Sumner** [comment]
Fixed in #10446
